### PR TITLE
file type, max and min document fields added and keys are refactored

### PR DESCRIPTION
**Pull request by simran2519**
file type, max and min document fields added and keys are refactored

### CODE REVIEW

File: api/src/main/java/com/community/api/endpoint/avisoft/controller/Customer/CustomerEndpoint.java [hunk 1]
```diff
@@ -47,6 +49,7 @@
 import java.lang.reflect.Field;
 import java.time.LocalDateTime;
 import java.util.*;
```
Comment (raman-kumar-avisoft) [inline api/src/main/java/com/community/api/endpoint/avisoft/controller/Customer/CustomerEndpoint.java]: dont import all classes use spefic classes only

File: api/src/main/java/com/community/api/endpoint/avisoft/controller/Customer/CustomerEndpoint.java [hunk 2]
```diff
@@ -11,13 +11,15 @@
 import com.community.api.entity.CustomCustomer;
 import com.community.api.entity.CustomerReferrer;
 import com.community.api.entity.CustomProduct;
+import com.community.api.entity.FileType;
 import com.community.api.services.*;
```
Comment (raman-kumar-avisoft) [inline api/src/main/java/com/community/api/endpoint/avisoft/controller/Customer/CustomerEndpoint.java]: don't use astric while importing classes

File: api/src/main/java/com/community/api/entity/FileType.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package com.community.api.entity;
+
+import com.community.api.utils.DocumentType;
+import com.fasterxml.jackson.annotation.JsonBackReference;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import javax.persistence.*;
```
Comment (raman-kumar-avisoft) [inline api/src/main/java/com/community/api/entity/FileType.java]: dont;s use astric while importing classes

File: api/src/main/java/com/community/api/services/DocumentStorageService.java [hunk 4]
```diff
@@ -203,25 +242,25 @@ public void saveAllDocumentTypes() {
...
+//
+//                new DocumentType(24,"PERSONAL_PHOTO", "A Personal Photgraph of SP"),
+//
+//
+//                new DocumentType(25, "CATEGORY", "The classification of individuals, such as gender categories: Male, Female, Other."),
+//                new DocumentType(26, "DISABILITY", "A term used to describe individuals with physical or mental impairments; 'person with a disability' is the preferred terminology."),
+//                new DocumentType(27, "EX-SERVICE-MEN", "An identification document required for veterans, typically used to access benefits or services."),
+//                new DocumentType(28, "NCC", "A document serving as proof of participation in the National Cadet Corps, often required for certain government applications."),
+//                new DocumentType(29, "SPORTS", "A personal photograph typically required for sports-related documentation, such as player registrations or team memberships."),
+//                new DocumentType(30, "FREEDOM FIGHTER", "A personal photograph required for identification and documentation purposes related to recognition and benefits for freedom fighters.")
```
Comment (raman-kumar-avisoft) [inline api/src/main/java/com/community/api/services/DocumentStorageService.java]: use multiline comments and while generating a PR try to remove commented code (can also add commented code as well but only neccesary).

File: api/src/main/java/com/community/api/utils/DocumentType.java [hunk 5]
```diff
@@ -1,11 +1,14 @@
 package com.community.api.utils;
 
+import com.community.api.entity.FileType;
+import com.fasterxml.jackson.annotation.JsonManagedReference;
 import lombok.AllArgsConstructor;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 import lombok.Setter;
 
 import javax.persistence.*;
```
Comment (raman-kumar-avisoft) [inline api/src/main/java/com/community/api/utils/DocumentType.java]: Don't use astrics while importing

File: api/src/main/java/com/community/api/utils/DocumentType.java [hunk 6]
```diff
@@ -22,8 +25,26 @@ public class DocumentType {
...
+
+    @Column(name = "max_document_size")
+    private String max_document_size;
+    @Column(name = "min_document_size")
+    private String min_document_size;
+//    @Column(name="max_document_size")
+//    private Integer max_document_size;
+//    @Column(name= "min_document_size")
+//    private Integer min_document_size;
+
```
Comment (raman-kumar-avisoft) [inline api/src/main/java/com/community/api/utils/DocumentType.java:48]: use multiline comments and try to remove the commented code

File: core/src/main/resources/runtime-properties/common-shared.properties [hunk 7]
```diff
@@ -253,18 +253,8 @@ database.password=1234
 database.driver=org.postgresql.Driver
 
 # this connection URL assumes that it is connecting to a schema called broadleaf
+database.url=jdbc:postgresql://192.168.0.138:5432/BroadleafTestConnection
 
-# Comment out the solr.source.* above and use the following
-
- database.url=jdbc:postgresql://192.168.0.138:5432/BroadleafTestConnection
-#database.url=jdbc:postgresql://localhost:5432/broadleaf
```
Comment (raman-kumar-avisoft) [inline core/src/main/resources/runtime-properties/common-shared.properties]: don' remove those line from the code they are for developer ease as we don't have to write those lines again if we want to use local db .

**Review by raman-kumar-avisoft — CHANGES_REQUESTED** [review]

**Review by kshtjvats — APPROVED** [review]

**Pull request merged**